doc/development/plugins.rst

```diff
@@ -232,6 +232,91 @@ or can include in-built transforms such as:
 * :func:`pennylane.devices.preprocess.validate_multiprocessing_workers`
 * :func:`pennylane.devices.preprocess.validate_adjoint_trainable_params`
 * :func:`pennylane.devices.preprocess.no_sampling`
+  
+**Decompositions:**
+
+The :func:`~pennylane.devices.preprocess.decompose` transform is typically required as part of the transform program that
+decomposes unsupported operations to the device's native gate set. To define this transform a stopping condition needs
+to be specified. This is a function mapping an operator to a boolean that determines whether the operator should be decomposed.
+
+For example, for a device supporting the ``CNOT`` gate, the stopping condition and the decompose transform
+can be specified like so:
+
+>>> from pennylane.devices.preprocess import decompose
+>>> def stopping_condition(op):
+...     return op.name == "CNOT"
```

Comment:
Maybe a more realistic example with a few more gates here? A device that only supports CNOT and nothing else looks strange.

Reply:
You can add `RZ` and `RY` into the gate set so that it looks more reasonable.